c/parallel/test/test_util.h

```diff
@@ -139,35 +141,43 @@ cccl_type_info get_type_info()
   info.size      = sizeof(T);
   info.alignment = alignof(T);
 
-  if constexpr (std::is_same_v<T, char>)
+  if constexpr (std::is_same_v<T, char> || (std::is_integral_v<T> && std::is_signed_v<T> && sizeof(T) == sizeof(char)))
   {
     info.type = cccl_type_enum::INT8;
   }
-  else if constexpr (std::is_same_v<T, uint8_t>)
+  else if constexpr (std::is_same_v<T, uint8_t>
+                     || (std::is_integral_v<T> && std::is_unsigned_v<T> && sizeof(T) == sizeof(char)
+                         && !std::is_same_v<T, bool>) )
   {
     info.type = cccl_type_enum::UINT8;
   }
-  else if constexpr (std::is_same_v<T, int16_t>)
+  else if constexpr (std::is_same_v<T, int16_t>
+                     || (std::is_integral_v<T> && std::is_signed_v<T> && sizeof(T) == sizeof(int16_t)))
```

Comment:
I understand the first two clauses as ways to handle `char`, `signed char` and `unsigned char`. But why are the other changes for 16-bit and higher types necessary? Which types should be additionally covered?

Reply:
Using `unsigned long long` used to throw an "Unsupported type" exception.